commit cd1eac832061bda4d6fd384c2048c64fc4067e3a
Author: Phil Elwell <phil@raspberrypi.org>
Date:   Fri Sep 29 10:32:19 2017 +0100

    amba_pl011: Insert mb() for correct FIFO handling
    
    The pl011 register accessor functions use the _relaxed versions of the
    standard readl() and writel() functions, meaning that there are no
    automatic memory barriers. When polling a FIFO status register to check
    for fullness, it is necessary to ensure that any outstanding writes have
    completed; otherwise the flags are effectively stale, making it possible
    that the next write is to a full FIFO.
    
    Signed-off-by: Phil Elwell <phil@raspberrypi.org>

diff --git a/drivers/tty/serial/amba-pl011.c b/drivers/tty/serial/amba-pl011.c
index 8b2250847551..47ab15a550e3 100644
--- a/drivers/tty/serial/amba-pl011.c
+++ b/drivers/tty/serial/amba-pl011.c
@@ -1387,6 +1387,7 @@ static bool pl011_tx_char(struct uart_amba_port *uap, unsigned char c,
 		return false; /* unable to transmit character */
 
 	pl011_write(c, uap, REG_DR);
+	mb();
 	uap->port.icount.tx++;
 
 	return true;
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1040_cd1eac832061bda4d6fd384c2048c64fc4067e3a.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1040_cd1eac832061bda4d6fd384c2048c64fc4067e3a.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
